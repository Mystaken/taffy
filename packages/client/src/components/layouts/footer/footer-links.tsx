import React, { FunctionComponent } from 'react';
import { useWiderThan } from '../../../hooks/media-query.hook';
import { Grid, Link } from '@material-ui/core';
import { FooterText } from './footer-text';

interface FooterLinkProps {
  name: string;
  href: string;
}

const FooterLink: FunctionComponent<FooterLinkProps> = ({ name, href }) => {
  return (
    <Grid item>
      <Link href={href}>
        <FooterText variant="subtitle2" align="center">
          {name}
        </FooterText>
      </Link>
    </Grid>
  );
};

export const FooterLinks: FunctionComponent = () => {
  const displayRow = useWiderThan(500);
  return (
    <Grid
      container
      direction={displayRow ? 'row' : 'column'}
      justify="center"
      alignItems="center"
      spacing={displayRow ? 2 : 1}>
      <FooterLink name="HOME" href="#"></FooterLink>
      <FooterLink name="OUR MEDIA" href="#"></FooterLink>
      <FooterLink name="ABOUT US" href="#"></FooterLink>
      <FooterLink name="FAQ" href="#"></FooterLink>
      <FooterLink name="TERMS AND SERVICES" href="#"></FooterLink>
    </Grid>
  );
};
