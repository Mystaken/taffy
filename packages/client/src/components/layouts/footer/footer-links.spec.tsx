import React from 'react';
import { create } from 'react-test-renderer';

import { FooterLinks } from './footer-links';

describe('<FooterLinks />', () => {
  it('has rendered correctly', () => {
    const component = create(<FooterLinks />);
    const tree = component.toJSON()!;
    expect(tree).toMatchSnapshot();
  });
});
