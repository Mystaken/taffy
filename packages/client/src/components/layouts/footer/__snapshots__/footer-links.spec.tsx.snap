// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FooterLinks /> has rendered correctly 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 MuiGrid-direction-xs-column MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
>
  <div
    className="MuiGrid-root MuiGrid-item"
  >
    <a
      className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
      href="#"
      onBlur={[Function]}
      onFocus={[Function]}
    >
      <h6
        className="MuiTypography-root MuiTypography-subtitle2 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
      >
        HOME
      </h6>
    </a>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item"
  >
    <a
      className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
      href="#"
      onBlur={[Function]}
      onFocus={[Function]}
    >
      <h6
        className="MuiTypography-root MuiTypography-subtitle2 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
      >
        OUR MEDIA
      </h6>
    </a>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item"
  >
    <a
      className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
      href="#"
      onBlur={[Function]}
      onFocus={[Function]}
    >
      <h6
        className="MuiTypography-root MuiTypography-subtitle2 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
      >
        ABOUT US
      </h6>
    </a>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item"
  >
    <a
      className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
      href="#"
      onBlur={[Function]}
      onFocus={[Function]}
    >
      <h6
        className="MuiTypography-root MuiTypography-subtitle2 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
      >
        FAQ
      </h6>
    </a>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item"
  >
    <a
      className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
      href="#"
      onBlur={[Function]}
      onFocus={[Function]}
    >
      <h6
        className="MuiTypography-root MuiTypography-subtitle2 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
      >
        TERMS AND SERVICES
      </h6>
    </a>
  </div>
</div>
`;
